lib/barbeque/message/job_execution.rb

```diff
@@ -7,6 +7,19 @@ class JobExecution < Base
       attr_reader :application # [String] To specify `job_definitions.application_id`
       attr_reader :job         # [String] To specify `job_definitions.name`
 
+      # Creates JobExecution messages from a notification which came from subscribed SNS topic
+      def self.create_message_from_sns_notification(raw_message, message_body, job_queue:)
```

Comment:
This introduces design inconsistency. `Barbeque::Message::JobExecution` is built for parsing a (SQS) message having `body['Type'] == 'JobExecution'`. But this parses a (SNS) message having `body['Type'] == 'Notification'`.

Can't we create `Barbeque::Message::Notification` and `Barbeque::MessageHandler::Notification` (and that creates `Barbeque::JobExecution` internally, inheriting `Barbeque::MessageHandler::JobExecution` if necessary)?

With such design, I believe that `Barbeque::Message::Notification` itself can have only `message_body['TopicArn']` and `message_body['Message']` (actual messages are only them, right?) and `Barbeque::MessageHandler::Notification` finds `SNSSubscription` from its `TopicArn` and `JobQueue` instance passed from `Barbeque::Worker`.

Actually message's `'Type'` field is created to have such design.